ipn/localapi/localapi_test.go

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+package localapi
+
+import (
+	"runtime"
+	"testing"
+
+	"tailscale.com/client/tailscale/apitype"
+)
+
+func TestValidHost(t *testing.T) {
+	valid := []string{
+		"",
+		apitype.LocalAPIHost,
+		"localhost:41112", // port is the safesocket port on Windows
+		"127.0.0.1:41112",
+		"[::1]:41112",
```

Comment:
But we don't use localhost TCP on Windows anymore. Do we even want to allow these now? cc @maisem

Reply:
yeah we don't need these anymore. Sent https://github.com/tailscale/tailscale/pull/7115 for cleanup

Reply:
so, on windows the validHost logic should be changed to reject anything other than the socket.

but we continue to use TCP elsewhere? this function is otherwise correct?

Reply:
i'll update this pr once https://github.com/tailscale/tailscale/pull/7115 is in

Reply:
PTAL